user-operator/src/main/java/io/strimzi/operator/user/operator/QuotasOperator.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator;
+
+import io.strimzi.api.kafka.model.KafkaUserQuotas;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.operator.resource.ReconcileResult;
+import io.strimzi.operator.user.model.QuotaUtils;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.quota.ClientQuotaAlteration;
+import org.apache.kafka.common.quota.ClientQuotaFilter;
+import org.apache.kafka.common.quota.ClientQuotaFilterComponent;
+import org.apache.kafka.common.quota.ClientQuotaEntity;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * KafkaUserQuotasOperator is responsible for managing quotas in Apache Kafka / Apache Zookeeper.
+ */
+public class QuotasOperator extends AbstractAdminApiOperator<KafkaUserQuotas, Set<String>> {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(QuotasOperator.class.getName());
+
+    /**
+     * Constructor
+     *
+     * @param vertx Vertx instance
+     * @param adminClient Kafka Admin client instance
+     */
+    public QuotasOperator(Vertx vertx, Admin adminClient) {
+        super(vertx, adminClient);
+    }
+
+    /**
+     * Reconciles Acl rules for given user
+     *
+     * @param reconciliation    The reconciliation
+     * @param username          User name of the reconciled user. When using TLS client auth, the username should be already in the Kafka format, e.g. CN=my-user
+     * @param desired           The desired quotas configuration
+     *
+     * @return the Future with reconcile result
+     */
+    @Override
+    public Future<ReconcileResult<KafkaUserQuotas>> reconcile(Reconciliation reconciliation, String username, KafkaUserQuotas desired) {
+        return getAsync(reconciliation, username)
+                .compose(current -> {
+                    if (desired == null) {
+                        if (current == null)    {
+                            LOGGER.debugCr(reconciliation, "No expected quotas and no existing quotas -> NoOp", username);
+                            return Future.succeededFuture(ReconcileResult.noop(null));
+                        } else {
+                            LOGGER.debugCr(reconciliation, "No expected quotas, but {} existing quotas -> Deleting quotas", username, current);
```

Comment:
Why do you use two parameters (`username, current`) but use only one placeholder? Applies also to logging below.